src/units/UnitsAnnotatedTypeFactory.java

```diff
@@ -433,6 +440,24 @@ public Void visitBinary(BinaryTree binaryTree, AnnotatedTypeMirror type) {
             return null;
         }
     }
+    
+    @Override
+    protected TypeAnnotator createTypeAnnotator() {
+        return new ListTypeAnnotator(
+                new UnitsDefaultForTypeAnnotator(this), super.createTypeAnnotator());
+    }
+    
+    protected class UnitsDefaultForTypeAnnotator extends DefaultForTypeAnnotator {
+        // Programmatically set the qualifier 
```

Comment:
Do you understand why these aren't just set declaratively at
https://github.com/opprop/units-inference/blob/master/src/units/qual/Dimensionless.java#L15 ?

Reply:
I believe this is because these defaults are only set for type checking mode and not inference mode.